### PR TITLE
feat(components/forms): improve file attachments error messaging for incorrect file types

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2955042](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2955042)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2553/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2553/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2553/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2553/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2553/playground/)

File: libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts [hunk 1]
```diff
@@ -73,6 +73,13 @@ export class SkyFileAttachmentComponent
   @Input()
   public acceptedTypes: string | undefined;
 
+  /**
+   * A custom error message to display when a file doesn't match the accepted types.
+   * This replaces a default error message that lists all accepted types.d
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts]: Trailing "d"

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.spec.ts [hunk 2]
```diff
@@ -474,7 +474,7 @@ describe('File drop component', () => {
 
     expect(typeError).toBeVisible();
     expect(typeError.textContent).toContain(
-      'Upload a file of type image/png, image/jpeg.',
+      'Upload a file of one of the following types: PNG, JPEG.',
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/file-drop.component.spec.ts]: This message doesn't match the text in the resources string template?

File: libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts]: fixed

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.spec.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-drop.component.spec.ts]: updated

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/2562

**Pull request merged**